wire-schema/src/main/resources/wire/extensions.proto

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package wire;
+
+import "google/protobuf/descriptor.proto";
+
+extend google.protobuf.FieldOptions {
+  // TODO(jwilson): reserve new numbers for these.
+
+  /**
+   * Annotates a field that starts to be available with a specified version.
+   *
+   * The value is the earliest version in which the annotated field is retained. It is pruned from
+   * all preceding versions.
+   */
+  optional int64 since = 22103;
+
+  /**
+   * Annotates a field that ceases to be available with a specified version.
+   *
+   * The value is the version that drops the annotated field. It is not available in that version or
+   * any subsequent version.
+   */
+  optional int64 until = 22104;
```

Comment:
is there a use case to support other types?  for example, api version codes that are represented as strings?

Reply:
Yeah, I think we should address this before we start using it.